<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>904</width>
    <height>437</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>数据合并</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>431</width>
      <height>411</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="4">
      <widget class="QLabel" name="docNameLabel">
       <property name="text">
        <string>请选择 .Docx 文件</string>
       </property>
      </widget>
     </item>
     <item row="1" column="6">
      <widget class="QToolButton" name="toolButton">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item row="2" column="6">
      <widget class="QToolButton" name="toolButton_3">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1" alignment="Qt::AlignRight">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Excel 数据</string>
       </property>
      </widget>
     </item>
     <item row="2" column="4">
      <widget class="QLabel" name="excelNameLabel">
       <property name="text">
        <string>请选择 .csv 文件</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4">
      <widget class="QTreeWidget" name="treeWidget">
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>300</height>
        </size>
       </property>
       <column>
        <property name="text">
         <string>模板变量</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>数据源</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="3" column="6">
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <widget class="QPushButton" name="pushButton_4">
         <property name="maximumSize">
          <size>
           <width>37</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>填充</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_2">
         <property name="maximumSize">
          <size>
           <width>37</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>测试</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="runButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="maximumSize">
          <size>
           <width>37</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>运行</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="1" alignment="Qt::AlignRight|Qt::AlignVCenter">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Word 模板</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1" alignment="Qt::AlignRight">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>规则</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1" alignment="Qt::AlignRight">
      <widget class="QLabel" name="label">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>25</height>
        </size>
       </property>
       <property name="text">
        <string>生成模式</string>
       </property>
      </widget>
     </item>
     <item row="4" column="4">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QRadioButton" name="radioButtonMultiple">
         <property name="text">
          <string>生成不同的文档</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radioButtonSingle">
         <property name="text">
          <string>合并为一个文档</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0" rowspan="4">
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="3" rowspan="4">
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>13</width>
         <height>86</height>
        </size>
       </property>
      </spacer>
     </item>
     <item row="1" column="5" rowspan="4">
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>13</width>
         <height>86</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
   <widget class="QTextBrowser" name="logBrowser">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>20</y>
      <width>431</width>
      <height>401</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:16px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:x-large; font-weight:600;&quot;&gt;Word数据合并&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#505050;&quot;&gt;操作说明：&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Courier New'; color:#505050;&quot;&gt;1. 将Word中需要更改的数据改为&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-weight:600; color:#505050;&quot;&gt;模板变量&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#505050;&quot;&gt;，模板变量就是用两个大括号&amp;quot;&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:12pt; font-weight:600; color:#505050;&quot;&gt;{{&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#505050;&quot;&gt;&amp;quot; &amp;quot;&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; font-size:12pt; font-weight:600; color:#505050;&quot;&gt;}}&lt;/span&gt;&lt;span style=&quot; font-family:'Courier New'; color:#505050;&quot;&gt;&amp;quot;包裹的变量，比如{{name}}，名称自定义，多个数据一样，可以用同一个名字。注意：Word需为docx后缀文件，如不是，请另存为.docx格式。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#505050;&quot;&gt;2. Excel另存为.csv文件，为了方便处理数据。注：如果需要处理的数据本身中含有英文逗号&amp;quot;&lt;/span&gt;&lt;span style=&quot; font-size:12pt; font-weight:600; color:#505050;&quot;&gt;,&lt;/span&gt;&lt;span style=&quot; color:#505050;&quot;&gt;&amp;quot;，则可能会有未知错误。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#505050;&quot;&gt;3. 选择两个文件，点【填充】按钮，填充对应的数据，点击【测试】，如果测试成功则可运行。否则，请修改数据。&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#505050;&quot;&gt;4. 生成模式：【默认】分为生成不同的文档（拆分为多个文件）；合并为一个文档（只生成一个文件），注意如果有特殊字体，此方法样式会使用基本字体，请注意样式问题。文件生成在运行目录下的 docxdata 文件夹内。&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;right&quot; style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#505050;&quot;&gt; © 姜小豆 &lt;/span&gt;&lt;span style=&quot; font-size:10pt; color:#505050;&quot;&gt;wsrfssx@yeah.net&lt;/span&gt;&lt;span style=&quot; color:#505050;&quot;&gt; 2019-1-30&lt;/span&gt;&lt;/p&gt;
&lt;hr /&gt;
&lt;p style=&quot; margin-top:12px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;运行日志&lt;/span&gt;：&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>20</y>
      <width>20</width>
      <height>401</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_open_word_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>455</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_test_rule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>330</y>
    </hint>
    <hint type="destinationlabel">
     <x>478</x>
     <y>293</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_open_excel_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_fill_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonMultiple</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_build_type()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>414</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>425</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slot_open_word_file()</slot>
  <slot>slot_open_excel_file()</slot>
  <slot>slot_test_rule()</slot>
  <slot>slot_run()</slot>
  <slot>slot_fill_data()</slot>
  <slot>slot_build_type()</slot>
 </slots>
</ui>
